### PR TITLE
feat: improve CCTP transaction steps

### PR DESCRIPTION
**Pull request by 0xArdy**

### CODE REVIEW

File: src/containers/Modals/Loading/GenericStep.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    txMetadata: { step: txStep, sourceHash },
+    transactionType,
+  } = useTransactionData();
+  const { fromChain } = useChain();
+
+  const sourceChainId = fromChain?.id;
+
+  return (
+    <>
+      {transactionType !== TransactionType.DEPOSIT &&
```
Comment (0xShaito) [inline src/containers/Modals/Loading/GenericStep.tsx:18]: Maybe ![DEPOSIT,WITHDRAW...].contains(transaction type) ?

File: src/containers/Modals/Loading/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }, [time, txStep]);
+
+  return (
+    <BaseModal type={ModalType.LOADING} title={t('MODALS.LOADING.title')}>
+      <SBox>
+        {/* Deposit flow */}
+        {transactionType === TransactionType.DEPOSIT && <DepositWithdrawStep />}
+
+        {/* Withdrawal flow */}
+        {transactionType === TransactionType.WITHDRAW && <DepositWithdrawStep />}
```
Comment (0xShaito) [inline src/containers/Modals/Loading/index.tsx:41]: So many repetitions of checks. Maybe we can create functions for it? isWithdraw(transactionType)?

A file with many of this fns

**Review by 0xShaito — CHANGES_REQUESTED** [review]

File: src/containers/Modals/Loading/GenericStep.tsx (hunk 1 above)
Comment (0xArdy) [reply src/containers/Modals/Loading/GenericStep.tsx:18]: my bad, i should have deleted these checks

File: src/containers/Modals/Loading/index.tsx (hunk 2 above)
Comment (0xArdy) [reply src/containers/Modals/Loading/index.tsx:41]: yeah i like that idea

File: src/containers/Modals/Loading/GenericStep.tsx (hunk 1 above)
Comment (0xArdy) [reply src/containers/Modals/Loading/GenericStep.tsx]: done https://github.com/defi-wonderland/superchain-bridge-ui/pull/82/commits/cb3a540e0996c217ea04e4e63d598127cea12f73

File: src/containers/Modals/Loading/index.tsx (hunk 2 above)
Comment (0xArdy) [reply src/containers/Modals/Loading/index.tsx]: done https://github.com/defi-wonderland/superchain-bridge-ui/pull/82/commits/ece11de8508116ab3a40e6a4b4f6dc191b86857b

**Review by 0xShaito — APPROVED** [review]

**Review by 0xtiti — APPROVED** [review]

**Pull request merged**